src/schemas/map.json

```diff
@@ -28,19 +28,89 @@
     "ItemsPath": {
       "$ref": "paths.json#/definitions/asl_ref_path"
     },
+    "Label": {
+      "type": "string"
+    },
     "MaxConcurrency": {
       "type": "number",
       "minimum": 0
     },
+    "ItemReader": {
+      "$comment": "need to improve this type",
+      "type": "object"
+    },
+    "ItemProcessor": {
+      "type": "object",
+      "allOf": [{ "$ref": "base-state-machine.json#" }],
+      "oneOf": [
+        {
+          "properties": {
+            "ProcessorConfig": {
+              "type": "object",
+              "properties": {
+                "Mode": {
+                  "type": "string",
+                  "enum": ["INLINE"]
+                }
+              }
+            }
+          }
+        },
+        {
+          "properties": {
+            "ProcessorConfig": {
+              "type": "object",
+              "properties": {
+                "Mode": {
+                  "type": "string",
+                  "enum": ["DISTRIBUTED"]
+                },
+                "ExecutionType": {
+                  "type": "string",
+                  "enum": ["EXPRESS", "STANDARD"]
+                }
+              }
+            }
+          }
+        }
+      ]
+    },
     "Iterator": {
-      "$ref": "state-machine.json#"
+      "$ref": "base-state-machine.json#"
     },
     "Parameters": {
       "$ref": "paths.json#/definitions/asl_payload_template"
     },
+    "ItemSelector": {
+      "$ref": "paths.json#/definitions/asl_payload_template"
+    },
+    "ItemBatcher":{
+      "type": "object",
+      "oneOf": [
+        {
+          "type": "object",
+          "properties": {
+            "MaxItemsPerBatch": {
+              "type": "number"
+            }
+          }
+        },
+        {
+          "type": "object",
+          "properties": {
+            "MaxItemsPerBatchPath": {
+              "$ref": "paths.json#/definitions/asl_ref_path"
+            }
+          }
+        }
+      ]
+    },
     "ResultSelector": {
       "$ref": "paths.json#/definitions/asl_payload_template"
     },
+    "ResultWriter": {
```

Comment:
need better typxingfor the ResultWriter as well. I think these fields are for the s3 bucket to write to.

Reply:
could you add a `$comment`, like you did for `ItemReader`?